### PR TITLE
feat(IATP): pluggable constraint to scope mapping

### LINKED ISSUE #3492: IATP: derive access scope from credential type
The consumer derives the scope required for a particular DSP request from the policy, and maps it onto a scope definition.

For example, a policy requiring a credential of type `MembershipCredential` would be translated into a scope `org.eclipse.edc.vc.type:MembershipCredential:read`.

For example, a policy constraint like this:
```json
{
  "constraint": {
    "leftOperand": "Membership", 
    "operator": "eq",
    "rightOperand": "active:1.0.0"
  }
}
```

would translate into `org.eclipse.edc.vc.type:MembershipCredential_1.0.0:read`

Requirements:
- converting the constraint into the scope string must be pluggable, there won't be a generic solution for EDC, because it makes no sense.
- the alias (= `org.eclipse.edc.vc.type`) must be configurable, as this is specific to the dataspace. This could be handled inside the constraint mapper.

documentation: 
- [access scopes](https://github.com/eclipse-tractusx/identity-trust/blob/8215853dddad222b1cdb81d2d16d202265ea698a/specifications/M1/verifiable.presentation.protocol.md#31-access-scopes)
- [conversion method](https://github.com/eclipse-tractusx/ssi-docu/blob/main/docs/architecture/policy_credentialtype_scope.md)

### PR DESCRIPTION
**Pull request by wolf4ood**
## What this PR changes/adds

Add pluggable constraints to scope mapping:

-  Introduces `ScopeExtractorRegistry` for registering custom extractors.
- Implements `IatpScopeExtractorFunction` and register it as pre-validator in the policy engine for invoking estractors in the registry and map the result in the `scope`

`ScopeExtractor` for now are not tied to a specific `scope` or `rule`.

## Why it does that

iatp adoption

## Linked Issue(s)

Closes #3492 

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/scope/ScopeExtractor.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * Performs the extraction of the scopes.
+     *
+     * @param leftValue  the left-side expression for the constraint.
+     * @param operator   the operator.
+     * @param rightValue the right-side expression for the constraint.
+     * @param context    the policy context
+     */
+    Set<String> extractScopes(Object leftValue, Operator operator, Object rightValue, PolicyContext context);
```
Comment (paullatzelsperger) [inline spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/scope/ScopeExtractor.java:39]: maybe use `Result<Set<String>>`?

File: extensions/common/iam/identity-trust/identity-trust-core/src/main/java/org/eclipse/edc/iam/identitytrust/core/IatpScopeExtractorExtension.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    public static final String CATALOG_REQUEST_SCOPE = "request.catalog";
+    public static final String NEGOTIATION_REQUEST_SCOPE = "request.contract.negotiation";
+    public static final String TRANSFER_PROCESS_REQUEST_SCOPE = "request.transfer.process";
+
+    @Inject
+    private PolicyEngine policyEngine;
+
+    @Inject
+    private ScopeExtractorRegistry scopeMapperRegistry;
```
Comment (paullatzelsperger) [inline extensions/common/iam/identity-trust/identity-trust-core/src/main/java/org/eclipse/edc/iam/identitytrust/core/IatpScopeExtractorExtension.java:41]: did you forget to add this extension to the `META-INF/services/...` file?

Comment (wolf4ood) [reply extensions/common/iam/identity-trust/identity-trust-core/src/main/java/org/eclipse/edc/iam/identitytrust/core/IatpScopeExtractorExtension.java:41]: Good catch :)

**Review by jimmarino — APPROVED** [review]

File: spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/scope/ScopeExtractor.java (hunk 1 above)
Comment (paullatzelsperger) [reply spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/scope/ScopeExtractor.java:39]: ah, right. makes sense 👍

**Review by paullatzelsperger — APPROVED** [review]

File: extensions/common/iam/identity-trust/identity-trust-core/src/main/java/org/eclipse/edc/iam/identitytrust/core/IatpScopeExtractorExtension.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import static org.eclipse.edc.iam.identitytrust.core.IatpScopeExtractorExtension.NAME;
+
+@Extension(NAME)
+public class IatpScopeExtractorExtension implements ServiceExtension {
+
+    public static final String NAME = "IATP scope extractor extension";
+
+    public static final String CATALOG_REQUEST_SCOPE = "request.catalog";
+    public static final String NEGOTIATION_REQUEST_SCOPE = "request.contract.negotiation";
+    public static final String TRANSFER_PROCESS_REQUEST_SCOPE = "request.transfer.process";
```
[the remark this answers is not in the document]
Comment (wolf4ood) [reply extensions/common/iam/identity-trust/identity-trust-core/src/main/java/org/eclipse/edc/iam/identitytrust/core/IatpScopeExtractorExtension.java:35]: Not sure but i think in this case not, we are not defining here additional scopes but just using them as constants. 
If i remember correctly they are defined as `@PolicyScope` in the DSP extensions

**Pull request merged**